### PR TITLE
Cormac mobilecommons connector

### PR DESCRIPTION
**Pull request by cmdelrio**
Thank you Daniel! Let me know if I can clarify anything.

### CODE REVIEW

File: parsons/mobilecommons/mobilecommons.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            # Append to final table
+            final_table.concat(response_table)
+            # Check to see if there are more pages and determine how many to retrieve
+            avail_pages = int(response_dict['response'][endpoint]['page_count'])
+            req_pages = math.ceil(limit/page_limit)
+            pages_to_get = avail_pages if avail_pages < req_pages else req_pages
+            # Go fetch other pages of data
+            i = 2
+            while i <= pages_to_get:
+                page_params = params + f'&page={str(i)}'
```
Comment (dannyboy15) [inline parsons/mobilecommons/mobilecommons.py:86]: same thing about using a dict for params (and about using ints in f-string) as above.

File: parsons/mobilecommons/mobilecommons.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                logger.info(f'Fetching rows {str(i*page_limit)} - {str((i+1)*page_limit)} '
+                            f'of {limit}')
```
Comment (dannyboy15) [inline parsons/mobilecommons/mobilecommons.py:88]: f-string implicitly convert ints to str

File: parsons/mobilecommons/mobilecommons.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                response = self.client.request(endpoint + self.companyid_param, 'GET',
+                                               params=page_params)
```
Comment (dannyboy15) [inline parsons/mobilecommons/mobilecommons.py:90]: same thing about params.

File: parsons/mobilecommons/mobilecommons.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                response_dict = xmltodict.parse(response.text, attr_prefix='', cdata_key='',
+                                                dict_constructor=dict)
+                response_table = Table(response_dict['response'][endpoint][data_key])
+                final_table.concat(response_table)
+                i += 1
```
Comment (dannyboy15) [inline parsons/mobilecommons/mobilecommons.py:95]: if you switch to for loop remove this.

File: parsons/mobilecommons/mobilecommons.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        # If initial response is bad, raise error
+        else:
```
Comment (dannyboy15) [inline parsons/mobilecommons/mobilecommons.py:97]: One other approach would be to check in if there's an error at the top and if not then fetch the data. It looks a little cleaner and keep you core logic from being indent more than it has to.

```python
if response.status_code != 200:
    # do error handling here
    ...
    raise

# do your data fetching here.
for i in range(2, pages_to_get + 1):
    ...
```

File: parsons/mobilecommons/mobilecommons.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        """
+        Retrieve broadcast data
+
+        `Args:`
+            :param start_time:
+            :param end_time:
+            :param status:
+            :param campaign_id:
+            :param limit:
+            :return:
```
Comment (dannyboy15) [inline parsons/mobilecommons/mobilecommons.py:117]: Try to be consistent with the docstring formatting.

File: parsons/mobilecommons/mobilecommons.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        """
+        # Still working on compiling the params, but this function does work atm with no params specified
+        params = ''
```
Comment (dannyboy15) [inline parsons/mobilecommons/mobilecommons.py:120]: if you use a dict for params, this should be fairly straightforward. you can also use the utility to parse dates.

```python
from parsons.utilities.datetime import parse_date

DATE_FMT = '%Y-%m-%d'
formatted_date = lambda x: parse_date(x).strftime(DATE_FMT)

...

params = {
    'start_time': formatted_date(start_time),
    'end_time': formatted_date(end_time),
}
```

**Review by dannyboy15 — COMMENTED** [review]
This looks really good! I left comments mostly to tidy up the code, and I didn't see any logic error. Let me know if you have questions or want to me to explain something further.

**Comment by sharinetmc** [comment]
Hi @cmdelrio! I just wanted to see where you were at making the changes danny suggested?

**Comment by cmdelrio** [comment]
@sharinetmc I think I actually incorporated all of these. This connector is basically ready, but I haven't had the chance to write unit tests.

**Review by shaunagm — CHANGES_REQUESTED** [review]
Just updating this so it has the correct Github status of "request changes", let me know how if/when you've got time to update this, happy to help however I can.

**Comment by shaunagm** [comment]
Closed in favor of #896

**Pull request closed without merging**